tmt/steps/provision/__init__.py

```diff
@@ -136,8 +136,14 @@ def go(self):
                         self.info('')
                 finally:
                     if isinstance(phase, ProvisionPlugin):
-                        if phase.guest():
-                            self._guests.append(phase.guest())
+                        guest = phase.guest()
```

Comment:
Another obvious note - ProvisionError is not logged nor printed - that has to be added too.

Reply:
I think that whole problem is that `finally:` block in tmt/steps/provision/__init__.py silences error and adds invalid guest to the store. 
My proposal is to log exception there (and it shouldn't be 'info' - this is error so _always_ reported to the user) by adding 
```
except (tmt.utils.RunError, tmt.utils.ProvisionError) as error:
  self.fail(error)
  raise
```
(RunError is raised by podman which is wrong. Clear solution would be to fix podman plugin to raise ProvisionError and ideally make that exception message more user friendly)

Reply:
File `tmt/steps/provision/__init__.py` is updated to catch `(tmt.utils.RunError, tmt.utils.ProvisionError)`. And file `tmt/steps/provision/testcloud.py` is updated as well.